app/controllers/hosts_controller.rb

```diff
@@ -90,6 +91,15 @@ def clone
     @host.valid?
   end
 
+  # Import a host from a compute resource
+  def import
```

Comment:
does it make sense to separate it to a ImportHostController ? i could see us implementing the create / destroy and maybe update rest verbs ( to add, remove and change a host to compute resource mapping). not to mention make the score of hosts controller better :)